lib/config/genconfig/awshaprovisionconfig.go

```diff
@@ -0,0 +1,1287 @@
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().Profile = profile
+	return
+}
+
+func (c *AwsHaProvisionConfig) PromptRegion() (err error) {
+	awsRegions := AwsRegionsImpFactory(c.Prompt)
+	awsRegion, err := awsRegions.Choose("AWS Region")
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().Region = awsRegion
+	return
+}
+
+func (c *AwsHaProvisionConfig) PromptVpcId() (err error) {
+	vpcId, err := c.Prompt.InputStringRequired("AWS VPC ID")
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().AwsVpcID = vpcId
+	return
+}
+
+func (c *AwsHaProvisionConfig) HasCidrBlockAddr() (isCiderBlock bool, err error) {
+	isCiderBlock, err = c.Prompt.Confirm("Do you want to use AWS CIDR Block", "yes", "no")
+	if err != nil {
+		return
+	}
+	if isCiderBlock {
+		err = c.PromptCidrBlockAddr()
+		if err != nil {
+			return
+		}
+	}
+	return
+}
+
+func (c *AwsHaProvisionConfig) PromptCidrBlockAddr() (err error) {
+	ciderBlockAddr, err := c.Prompt.InputStringRegex("AWS CIDR Block Address", IP_REGEX)
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().AwsCidrBlockAddr = ciderBlockAddr
+	return
+}
+
+func (c *AwsHaProvisionConfig) PromptPrivateSubnet() (err error) {
+	privateSubnet1, err := c.Prompt.InputStringRequired("AWS Private Subnet 1:")
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().PrivateCustomSubnets = []string{privateSubnet1}
+
+	privateSubnet2, err := c.Prompt.InputStringRequired("AWS Private Subnet 2:")
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().PrivateCustomSubnets = append(c.Config.InitAws().InitConfigAwsSettings().PrivateCustomSubnets, privateSubnet2)
+
+	privateSubnet3, err := c.Prompt.InputStringRequired("AWS Private Subnet 3:")
+	if err != nil {
+		return
+	}
+	c.Config.InitAws().InitConfigAwsSettings().PrivateCustomSubnets = append(c.Config.InitAws().InitConfigAwsSettings().PrivateCustomSubnets, privateSubnet3)
+
+	return
+}
+
+func (c *AwsHaProvisionConfig) PromptPublicSubnet() (err error) {
+	hasPublicSubnets, err := c.Prompt.Confirm("Do you have Public Subnets", "yes", "no")
+	if err != nil {
+		return
+	}
+	if hasPublicSubnets {
+		publicSubnet1, err1 := c.Prompt.InputStringRequired("AWS Public Subnet 1:")
+		if err1 != nil {
+			return err1
+		}
+		c.Config.InitAws().InitConfigAwsSettings().PublicCustomSubnets = []string{publicSubnet1}
+
+		publicSubnet2, err1 := c.Prompt.InputStringRequired("AWS Public Subnet 2:")
+		if err1 != nil {
+			return err1
+		}
+		c.Config.InitAws().InitConfigAwsSettings().PublicCustomSubnets = append(c.Config.InitAws().InitConfigAwsSettings().PublicCustomSubnets, publicSubnet2)
+
+		publicSubnet3, err1 := c.Prompt.InputStringRequired("AWS Public Subnet 3:")
+		if err1 != nil {
+			return err1
+		}
+		c.Config.InitAws().InitConfigAwsSettings().PublicCustomSubnets = append(c.Config.InitAws().InitConfigAwsSettings().PublicCustomSubnets, publicSubnet3)
```

Comment:
We can add checks if all public or private subnets in array are unique

Reply:
I will consider this as feature enhancement for next time.